debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbStreamingChangeEventSource.java

```diff
@@ -299,16 +300,16 @@ protected MongoDbOffsetContext initializeOffsets(MongoDbConnectorConfig connecto
         final Map<ReplicaSet, BsonDocument> positions = new LinkedHashMap<>();
         replicaSets.onEachReplicaSet(replicaSet -> {
             LOGGER.info("Determine Snapshot Offset for replica-set {}", replicaSet.replicaSetName());
-            MongoPrimary primaryClient = establishConnectionToPrimary(partition, replicaSet);
-            if (primaryClient != null) {
+            MongoPreferredNode client = establishConnectionToPreferred(partition, replicaSet, ReadPreference.primaryPreferred());
```

Comment:
Should we be calling such references "node" or "client"? I think we are reserving "client" for MongoClient elsewhere.

Reply:
Yea the naming here is a bit flaky.. However it is a wrapper client essentially. I will try to go through the names one more time and put some consistency in.

Reply:
Variables/parameters/attributes of ``MongoPreferredNode`` type are now named simply ``mongo`` whereas the actual clients (e.g.  lambda parameters of ``MongoPreferredNode#execute``) are named ``client``.